File: src/main/scala/com/github/jarol/azure/search/spark/sql/connector/schema/conversion/InferSchemaRule.scala
package com.github.jarol.azure.search.spark.sql.connector.schema.conversion

/**
 * Conversion rules to be used for schema inference
 */

trait InferSchemaRule
  extends SearchSparkConversionRule {

  override final def useForSchemaInference(): Boolean = true

  override def useForSchemaConversion(): Boolean = false
}

File: src/main/java/com/github/jarol/azure/search/spark/sql/connector/schema/conversion/InferSchemaRule.java
package com.github.jarol.azure.search.spark.sql.connector.schema.conversion;

/**
 * Conversion rule to use for schema inference
 */

public abstract class InferSchemaRule
        implements SearchSparkConversionRule {

    @Override
    public final boolean useForSchemaInference() {
        return true;
    }

    @Override
    public final boolean useForSchemaConversion() {
        return false;
    }
}
